kayak_core/src/styles/mod.rs

```diff
@@ -141,23 +155,91 @@ macro_rules! define_styles {
 }
 
 define_styles! {
+
+    /// A struct used to define the look of a widget
+    ///
+    /// # Example
+    ///
+    /// All fields are `pub`, so you can simply define your styles like:
+    ///
+    /// ```
+    /// # use kayak_core::styles::{Style, StyleProp, Units};
+    /// let style = Style {
+    ///   width: StyleProp::Value(Units::Pixels(100.0)),
+    ///   // Or using `into()` to convert to `StyleProp`
+    ///   // width: Units::Pixels(100.0).into(),
+    ///   ..Default::default()
+    /// };
+    /// ```
+    ///
+    /// You can also create styles using other styles in a builder-like syntax:
+    ///
+    /// ```
+    /// # use kayak_core::styles::{Style, StyleProp, Units};
+    /// let style_a = Style {
+    ///   width: Units::Pixels(100.0).into(),
+    ///   ..Default::default()
+    /// };
+    /// let style_b = Style {
+    ///   height: Units::Pixels(100.0).into(),
+    ///   ..Default::default()
+    /// };
+    ///
+    /// let style = Style::default() // <- Initializes all fields as `StyleProp::Unset`
+    ///   // Applied first (sets any `StyleProp::Unset` fields)
+    ///   .with_style(&style_a)
+    ///   // Applied second (sets any remaining `StyleProp::Unset` fields)
+    ///   .with_style(&style_b);
+    /// ```
     #[derive(Debug, Default, Clone, PartialEq)]
     pub struct Style {
+        /// The background color of this widget
+        ///
+        /// Only applies to widgets marked [`RenderCommand::Quad`]
         pub background_color : StyleProp<Color>,
+        /// The color of the border around this widget
+        ///
+        /// Currently, this controls all border sides.
+        ///
+        /// Only applies to widgets marked [`RenderCommand::Quad`]
         pub border_color: StyleProp<Color>,
+        /// The radius of the corners (in pixels)
+        ///
+        /// The order is (Top, Right, Bottom, Left).
+        ///
+        /// Only applies to widgets marked [`RenderCommand::Quad`] and [`RenderCommand::Image`]
```

Comment:
Ideally we should allow both quads and images to have borders and background colors, but this is accurate right now. 👍